test/python/circuit/library/test_evolution_gate.py

```diff
@@ -113,6 +113,29 @@ def test_suzuki_trotter_manual(self):
 
         self.assertEqual(evo_gate.definition.decompose(), expected)
 
+    @idata(
+        [
+            [X + Y, 0.5, 1],
+            [X, 0.238, 2],
+        ]
+    )
+    @unpack
+    def test_qdrift_manual(self, op, time, reps):
+        """Test the evolution circuit of Suzuki Trotter against a manually constructed circuit."""
+        qdrift = QDrift(reps=reps)
+        evo_gate = PauliEvolutionGate(op, time, synthesis=qdrift)
+        evo_gate.definition.decompose()
+
+        # manually construct expected evolution
+        expected = QuantumCircuit(1)
+        for pauli in qdrift.sampled_ops:
+            if pauli[0].to_label() == "X":
+                expected.rx(2 * pauli[1], 0)
+            elif pauli[0].to_label() == "Y":
+                expected.ry(2 * pauli[1], 0)
+
+        self.assertEqual(evo_gate.definition.decompose(), expected)
```

Comment:
I saw randomness in the definition of the product formula.  Is this test guaranteed to produce exactly the same result every single time?

Reply:
no and it is not supposed to do so (see my above comment regarding how the algorithm works). That's why the manual circuit is built from `qdrift.sampled_ops`, which are those operations that were sampled and therefore should be in the circuit.

Reply:
It's fine if the algorithm itself involves randomness, but having randomness in tests isn't a good idea because it often causes stochastic test failures on unrelated PRs and makes the test suite flaky.  A random algorithm needs to have some way of seeding itself (so users can reproduce data runs if necessary), and all tests should set that seed to something fixed, so they test the same paths every time.

Reply:
I've added a seed at the beginning of the tests.